flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/windowing/WindowOperator.java

```diff
@@ -774,6 +774,14 @@ public KeyedStateStore windowState() {
 		public KeyedStateStore globalState() {
 			return WindowOperator.this.getKeyedStateStore();
 		}
+
+		public <X> void output(OutputTag<X> outputTag, X value) {
+			if (outputTag == null) {
+				throw new IllegalArgumentException("OutputTag must not be null.");
+			}
+			output.collect(outputTag, new StreamRecord<>(value, cleanupTime(window)));
```

Comment:
I think this should be `window.maxTimestamp()` to match `emitWindowContents()`.

Reply:
Make sense. Fixed